### ISSUE TITLE
Add "viewed" count to logs

### ISSUE THREAD

**Issue by rpurdie** [buildbot/buildbot#4895]
Its hard to come up with a policy about how long things should be kept for in the database. We (as in the Yocto Project) would love to be able to define a retention policy of only keeping logs for:

a) builds which had failures (successes aren't interesting)
b) builds where someone viewed the logs

Would it be possible to track the information for b) in the database so we could know which ones are better candidates for removal?

**Comment by p12tic** [comment]
I think this would be useful feature for most large setups. Not sure yet about the current capabilities.

**Comment by tardyp** [comment]
so if you viewed the log, at least once, then it is important, and we should keep it for a little while more?

**Comment by rpurdie** [comment]
Yes, we only ever view a very small subset of the logs and if we viewed it, it therefore has more importance and is worth keeping for longer.
